assemblyline/datastore/helper.py

```diff
@@ -1054,9 +1053,12 @@ def list_all_heuristics(self, as_obj=True) -> Union[List[dict], List[Heuristic]]
     @elasticapm.capture_span(span_type='datastore')
     def save_or_freshen_file(self, sha256, fileinfo, expiry, classification,
                              cl_engine=forge.get_classification(), redis=None, is_section_image=False):
-        with Lock(f'save-or-freshen-file-{sha256}', 5, host=redis):
-            current_fileinfo = self.ds.file.get_if_exists(
-                sha256, as_obj=False, force_archive_access=config.datastore.ilm.update_archive) or {}
+        while True:
```

Comment:
What if the line `self.ds.file.save(sha256, current_fileinfo, version=version)` always errors? Wouldn't that create an infinite loop?

Reply:
Only if errors with version conflict exception each time. Other exceptions will generate a stack trace. 

The VersionConflictException only happens when you are trying to save a document but someone beats you to it. So it loops again getting a new sequence number to save the document and try again... That infinite loop is basically by desing.